mythtv/programs/mythfrontend/themechooser.cpp

```diff
@@ -167,55 +167,32 @@ void ThemeChooser::Load(void)
-    QString MythVersion(GetMythSourceVersion());
-    static const QRegularExpression trunkver{"\\Av[0-9]+-pre.*\\z", QRegularExpression::CaseInsensitiveOption};
-    static const QRegularExpression validver{
-        "\\Av[0-9]+.*\\z", QRegularExpression::CaseInsensitiveOption};
-
-    auto match = validver.match(MythVersion);
-    if (!match.hasMatch())
-    {
-        LOG(VB_GENERAL, LOG_ERR, QString("Invalid MythTV version %1, will use themes from trunk").arg(MythVersion));
-        MythVersion = "trunk";
-    }
-    match = trunkver.match(MythVersion);
-    if (match.hasMatch())
-        MythVersion = "trunk";
+    uint8_t major { 0 };
+    uint8_t minor { 0 };
+    bool    devel { false };
+    bool    parsed = ParseMythSourceVersion(devel, major, minor);
 
-    if (MythVersion == "trunk")
+    if (!parsed || devel)
     {
-        LoadVersion(MythVersion, themesSeen, true);
-        LOG(VB_GUI, LOG_INFO, QString("Loading themes for %1").arg(MythVersion));
+        if (!parsed)
+            LOG(VB_GENERAL, LOG_ERR,
+                QString("Invalid MythTV version %1, will use themes from trunk")
+                .arg(GetMythSourceVersion()));
+        LOG(VB_GUI, LOG_INFO, QString("Loading themes for trunk"));
+        LoadVersion("trunk", themesSeen, true);
     }
     else
     {
-        MythVersion = MYTH_BINARY_VERSION; // Example: 29.20161017-1
-        // Remove the date part and the rest, eg 29.20161017-1 -> 29
-        MythVersion.remove(kVersionDateRE);
+        QString MythVersion { QString::number(major) };
         LOG(VB_GUI, LOG_INFO, QString("Loading themes for %1").arg(MythVersion));
         if (LoadVersion(MythVersion, themesSeen, true))
         {
-
-            // If a version of the theme for this tag exists, use it...
-            // MYTH_SOURCE_VERSION - examples v29-pre-574-g92517f5, v29-Pre, v29.1-21-ge26a33c
-            static const QRegularExpression subexp{"v[0-9]+\\.([0-9]+)-*", QRegularExpression::CaseInsensitiveOption};
-            // This captures the subversion, i.e. the number after a dot
-            match = subexp.match(GetMythSourceVersion());
-            if (match.hasMatch())
+            for (int idx = minor ; idx > 0; idx--)
             {
                 QString subversion;
-                LOG(VB_GUI, LOG_INFO, QString("Loading version %1").arg(match.capturedView(1).toInt()));
-                for (int idx = match.capturedView(1).toInt(); idx > 0; --idx)
-                {
-                    subversion = MythVersion + "." + QString::number(idx);
-                    LOG(VB_GUI, LOG_INFO, QString("Loading themes for %1").arg(subversion));
-                    LoadVersion(subversion, themesSeen, false);
-                }
-            }
-            else
-            {
-                LOG(VB_GENERAL, LOG_WARNING, QString("Failed to match theme for %1").arg(GetMythSourceVersion()));
+                subversion = MythVersion + "." + QString::number(idx);
```

Comment:
`QString majmin = QString("%1.%2").arg(major).arg(i);` to match with the backend code?

Reply:
Either would work. I was minimizing changes to this function. It doesn't look like it from this diff, but with the "-b" argument this line and the two following are unchanged.